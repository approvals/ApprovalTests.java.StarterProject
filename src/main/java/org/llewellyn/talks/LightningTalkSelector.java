package org.llewellyn.talks;

import java.awt.BorderLayout;
import java.awt.Dimension;
import java.awt.Font;

import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JPanel;

import org.random.selector.RamdomSelector;

import com.spun.util.ObjectUtils;
import com.spun.util.WindowUtils;

public class LightningTalkSelector extends JPanel
{
  private JLabel               commitLabel;
  private JButton              nextButton;
  private RamdomSelector<Talk> talks;
  public LightningTalkSelector()
  {
    talks = new RamdomSelector<>(Talk.class, LightningTalks.getLlewellyns());
    setLayout();
  }
  private void setLayout()
  {
    this.setLayout(new BorderLayout());
    this.setPreferredSize(new Dimension(1200, 75));
    commitLabel = new JLabel("Up Next : ");
    Font font = new Font("Courier", Font.BOLD, 28);
    commitLabel.setFont(font);
    this.add(commitLabel, BorderLayout.WEST);
    nextButton = new JButton("Next");
    nextButton.addActionListener(e -> next());
    this.add(nextButton, BorderLayout.EAST);
  }
  private Object next()
  {
    Talk talk = talks.get();
    commitLabel.setText("Up Next : " + talk.getTitle());
    this.invalidate();
    ObjectUtils.throwAsError(() -> Runtime.getRuntime().exec("open " + talk.getPath()));
    return null;
  }
  public static void main(String[] args)
  {
    WindowUtils.testPanel(new LightningTalkSelector());
  }
}
