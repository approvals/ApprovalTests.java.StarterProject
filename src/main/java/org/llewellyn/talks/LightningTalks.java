package org.llewellyn.talks;

import java.io.IOException;

import org.random.selector.RamdomSelector;

import com.spun.util.tests.TestUtils;

public class LightningTalks
{
  public static Talk _(String title, String path)
  {
    return new Talk(title, path);
  }
  public static Talk[] getLlewellyns()
  {
    return new Talk[]{_("Test Driven Math", "https://www.desmos.com/"),
                      _(" 10 X", "/Users/llewellyn/Documents/10x.pptx"),
                      _(" A swimming pool isn’t just a bigger bathtub",
                          "/Users/llewellyn/Documents/Infographics/A Swimming Pool is not just a bigger bathub.pptx"),
                      _(" Bdd vs TDD", "/Users/llewellyn/Documents/Bdd vs TDD.pptx"),
                      _(" Arlo’s Git Notation", "/Users/llewellyn/Documents/Belshee Git Notation.pptx"), // need improvement
                      _(" The Curse of knowledge",
                          "/Users/llewellyn/Documents/The Curse of Knowledge - parts.pptx"),
                      _(" Do NOT use the greater than sign in programming",
                          "/Users/llewellyn/Documents/Do NOT use the greater than sign in programming 16x9.pptx"),
                      _(" DocDoD", "/Users/llewellyn/Documents/DocDoD.pptx"),
                      _(" Sparrows", "/Users/llewellyn/Github/RefactoringClass/Slides/Sparrows_SongAndHouse.pptx"),
                      _(" Sticky Notes", "/Users/llewellyn/Github/RefactoringClass/Slides/Sticky Notes.pptx"),
                      _(" Leveling up", "/Users/llewellyn/Documents/Leveling Up.pptx"),
                      _(" On being the best",
                          "/Users/llewellyn/Library/Mobile Documents/com~apple~Keynote/Documents/On Being the Best Animation.key"),
                      _(" Mob Programming the best of the team",
                          "/Users/llewellyn/Documents/Infographics/Mob Programming Infographic.pptx"),
                      _(" Quantum Computing",
                          "/Users/llewellyn/Llewellyn Falco/Presentations/Quantum Computing - 2019.pptx"),
                      _(" Theory Based thread testing", "/Users/llewellyn/Documents/Thread base theory test.pptx"),
                      _(" Better Lunches", "/Users/llewellyn/Documents/Infographics/Better Lunches.pptx"),
                      _(" Generic Type Information at Runtime in Java",
                          "/Users/llewellyn/Documents/Generic Runtime Information in Java.pptx"),
                      _(" Sustainable Pace", "/Users/llewellyn/Documents/Sustainable Pace.pptx"),
                      _(" Decision trees", "/Users/llewellyn/Documents/Decision trees.pptx")};
  }
  public static void main(String[] args) throws IOException
  {
    RamdomSelector<Talk> talks = new RamdomSelector<>(Talk.class, getLlewellyns());
    while (!talks.isEmpty())
    {
      Talk talk = talks.get();
      System.out.print("Next up: " + talk.getTitle());
      TestUtils.displayFile(talk.getPath());
      System.in.read();
    }
    System.out.println("That's all folks!");
  }
}
