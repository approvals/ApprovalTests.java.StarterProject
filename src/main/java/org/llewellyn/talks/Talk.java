package org.llewellyn.talks;

public class Talk
{
  private String title;
  private String path;
  public Talk(String title, String path)
  {
    this.title = title;
    this.path = path;
  }
  public String getTitle()
  {
    return title;
  }
  public String getPath()
  {
    return path;
  }
}
