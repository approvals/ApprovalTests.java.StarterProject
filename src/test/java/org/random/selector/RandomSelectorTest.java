package org.random.selector;

import java.io.File;
import java.util.Random;
import java.util.stream.Stream;

import org.approvaltests.Approvals;
import org.approvaltests.legacycode.Range;
import org.junit.Assert;
import org.junit.Test;
import org.llewellyn.talks.LightningTalks;
import org.llewellyn.talks.Talk;

public class RandomSelectorTest
{
  @Test
  public void test()
  {
    RamdomSelector<Integer> selector = new RamdomSelector<>(Integer.class, Range.get(1, 10));
    selector.setRandom(new Random(10));
    Approvals.verifyAll("Selected", Stream.of(Range.get(1, 10)).map(__ -> selector.get()).toArray());
  }
  @Test
  public void testEmpty()
  {
    RamdomSelector<Integer> selector = new RamdomSelector<>(Integer.class, Range.get(1, 2));
    selector.get();
    selector.get();
    Approvals.verifyException(() -> selector.get());
  }
  @Test
  public void testTalks()
  {
    Talk[] talks = LightningTalks.getLlewellyns();
    System.out.println(String.format("Testing %s talks", talks.length));
    for (Talk talk : talks)
    {
      String message = String.format("%s:\n %s", talk.getTitle(), talk.getPath());
      if (talk.getPath().startsWith("http"))
      {
        continue;
      }
      Assert.assertTrue(message, new File(talk.getPath()).exists());
      if (talk.getPath().contains(" "))
      {
        File oldFile = new File(talk.getPath());
        File newFile = new File(talk.getPath().replace(" ", "_"));
        oldFile.renameTo(newFile);
        System.out.println(newFile);
      }
      //Assert.assertTrue(message, !talk.getPath().contains(" "));
    }
  }
}
