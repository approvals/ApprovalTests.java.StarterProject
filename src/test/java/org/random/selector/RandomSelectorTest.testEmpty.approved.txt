java.lang.RuntimeException: All [2] items have been picked, there are no more remaining.
